File: MaWPictures/src/main/java/us/mikeandwan/photos/prefs/PhotoDisplayPreference.java
package us.mikeandwan.photos.prefs;

import android.content.SharedPreferences;


public class PhotoDisplayPreference {
    private final SharedPreferences _sharedPrefs;


    public PhotoDisplayPreference(SharedPreferences sharedPrefs) {
        _sharedPrefs = sharedPrefs;
    }


    public boolean getDoDisplayPhotoToolbar() {
        return _sharedPrefs.getBoolean("display_toolbar", true);
    }


    public boolean getDoDisplayThumbnails() {
        return _sharedPrefs.getBoolean("display_thumbnails", true);
    }


    public boolean getDoDisplayTopToolbar() {
        return _sharedPrefs.getBoolean("display_top_toolbar", true);
    }


    public boolean getDoFadeControls() {
        return _sharedPrefs.getBoolean("fade_controls", true);
    }


    public int getSlideshowIntervalInSeconds() {
        return Integer.parseInt(_sharedPrefs.getString("slideshow_interval", "3"));
    }
}

File: MaWPhotos/src/main/java/us/mikeandwan/photos/preferences/PhotoDisplayPreference.kt
package us.mikeandwan.photos.preferences

import android.content.SharedPreferences

class PhotoDisplayPreference(private val _sharedPrefs: SharedPreferences) {
    val doDisplayPhotoToolbar: Boolean
        get() = _sharedPrefs.getBoolean("display_toolbar", true)
    val doDisplayThumbnails: Boolean
        get() = _sharedPrefs.getBoolean("display_thumbnails", true)
    val doDisplayTopToolbar: Boolean
        get() = _sharedPrefs.getBoolean("display_top_toolbar", true)
    val doFadeControls: Boolean
        get() = _sharedPrefs.getBoolean("fade_controls", true)
    val slideshowIntervalInSeconds: Int
        get() = _sharedPrefs.getString("slideshow_interval", "3")!!.toInt()
}